Name threads in caffe2/torch/distributed/checkpoint AsyncCheckpointExecutor

dsesh opened the pull request:
Differential Revision: D78493333




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

d4l3k reviewed (approved):
LGTM

Do you also want to set the pthreads name for quickstack/gdb/below?

Ex: https://github.com/pytorch/pytorch/blob/7b72e5b3ad989d02802909b64f322b2b7b69913b/torch/utils/data/_utils/pin_memory.py#L23

dsesh committed:
Name threads in caffe2/torch/distributed/checkpoint AsyncCheckpointExecutor (#158612)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/158612

Test Plan:
Built `training_platform.worker:c44710f0f91f8b4ed58f517864a7ba8b`, and ran [`f765449601-TrainingApplication_S3BTR`](https://www.internalfb.com/mlhub/pipelines/runs/mast/f765449601-TrainingApplication_S3BTR?version=0&tab=summary&env=PRODUCTION).

[SBDive profile](https://www.internalfb.com/intern/sbdive/?id=f765449601-TrainingApplication_S3BTR-84eceeca-8b03-410b-9e39-3840ec9cf185) of the job shows  threads named `AsyncCheckpointExecutor`

All of rank0's Python threads
{F1980370763}

All `AsyncCheckpointExecutor` across ranks
 {F1980371593}

Rollback Plan:

Reviewed By: d4l3k

Differential Revision: D78493333
